### PR TITLE
feat(storage): store all column descs in cell based table

### PR DESCRIPTION
**Pull request by BugenZhao**
## What's changed and what's your intention?

This PR stores all column descs in cell based table and add another field of `output_ids` to control which column to ser/de.
This is a prerequisite of supporting point get or range scan with vnode encoded in the key.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- #3316

### CODE REVIEW

File: src/storage/src/table/cell_based_table.rs [hunk 1]
```diff
@@ -84,24 +87,48 @@ fn err(rw: impl Into<RwError>) -> StorageError {
 }
 
 impl<S: StateStore> CellBasedTable<S> {
+    /// Create a [`CellBasedTable`] with given a complete set of `columns`.
```
Comment (xx01cyx) [inline src/storage/src/table/cell_based_table.rs:90]:
```suggestion
    /// Create a [`CellBasedTable`] given a complete set of `columns`.
```

File: src/storage/src/table/cell_based_table.rs [hunk 2]
```diff
@@ -84,24 +87,48 @@ fn err(rw: impl Into<RwError>) -> StorageError {
...
+            keyspace,
+            columns,
+            column_ids,
+            order_types,
+            pk_indices,
+            dist_key_indices,
+        )
+    }
+
+    /// Create a [`CellBasedTable`] with given a complete set of `columns` and a partial set of
```
Comment (xx01cyx) [inline src/storage/src/table/cell_based_table.rs:110]:
```suggestion
    /// Create a [`CellBasedTable`] given a complete set of `columns` and a partial set of
```

File: proto/plan_common.proto [hunk 3]
```diff
@@ -52,7 +52,8 @@ message ColumnCatalog {
 
 message CellBasedTableDesc {
   uint32 table_id = 1;
-  repeated OrderedColumnDesc order_key = 2;
+  repeated ColumnDesc columns = 2;
+  repeated OrderedColumnDesc order_key = 3;
```
Comment (skyzh) [inline proto/plan_common.proto:56]: I think we should also store pk and distribution key here?

File: src/storage/src/table/cell_based_table.rs [hunk 4]
```diff
@@ -49,33 +47,38 @@ pub struct CellBasedTable<S: StateStore> {
...
 
-    column_ids: Vec<ColumnId>,
+    /// Indices of primary keys.
+    /// Note that the index is based on the all columns of the table, instead of the output ones.
+    // FIXME: revisit constructions and usages.
+    pk_indices: Vec<usize>,
 
-    /// Indices of distribution keys in full row for computing value meta. None if value meta is
-    /// not required.
+    /// Indices of distribution keys for computing value meta. None if value meta is not required.
```
Comment (xx01cyx) [inline src/storage/src/table/cell_based_table.rs:73]:
```suggestion
    /// Indices of distribution keys for computing vnode. None if vnode falls to default value.
```

File: src/batch/src/executor/row_seq_scan.rs [hunk 5]
```diff
@@ -138,25 +139,52 @@ impl BoxedExecutorBuilder for RowSeqScanExecutorBuilder {
...
+            .iter()
+            .copied()
+            .map(ColumnId::from)
+            .collect();
+
+        let pk_descs: Vec<OrderedColumnDesc> =
+            table_desc.order_key.iter().map(|d| d.into()).collect();
         let order_types: Vec<OrderType> = pk_descs.iter().map(|desc| desc.order).collect();
 
+        let pk_indices = pk_descs
```
Comment (wcy-fdu) [inline src/batch/src/executor/row_seq_scan.rs:162]: Can we pass pk_indices from frontend? just like top_n or join.

File: src/storage/src/table/cell_based_table.rs [hunk 6]
```diff
@@ -49,33 +47,38 @@ pub struct CellBasedTable<S: StateStore> {
...
+    /// Note that the index is based on the all columns of the table, instead of the output ones.
+    // FIXME: revisit constructions and usages.
     dist_key_indices: Option<Vec<usize>>,
```
Comment (xx01cyx) [inline src/storage/src/table/cell_based_table.rs:76]: Which is better, `Option<Vec<usize>>` or `Vec<usize>`? 🤔

File: proto/plan_common.proto (hunk 3 above)
Comment (BugenZhao) [reply proto/plan_common.proto:56]: You are very forward-looking! Plan to do this in next PRs. cc @st1page

File: src/batch/src/executor/row_seq_scan.rs (hunk 5 above)
Comment (BugenZhao) [reply src/batch/src/executor/row_seq_scan.rs:162]: You are very forward-looking! Plan to do this in next PRs. cc @st1page

File: src/storage/src/table/cell_based_table.rs (hunk 6 above)
Comment (BugenZhao) [reply src/storage/src/table/cell_based_table.rs:76]: Good catch. I think both are acceptable.

**Review by wcy-fdu — APPROVED** [review]
LGTM!
Will the next PR change Vnode encoding?

File: src/batch/src/executor/row_seq_scan.rs (hunk 5 above)
Comment (skyzh) [reply src/batch/src/executor/row_seq_scan.rs:162]: 🥵🥵 foward 🥵🥵 looking!

**Pull request merged**